lib/AST/ASTImporter.cpp

```diff
@@ -302,8 +264,18 @@ namespace clang {
     void InitializeImportedDecl(Decl *FromD, Decl *ToD) {
       ToD->IdentifierNamespace = FromD->IdentifierNamespace;
       if (FromD->hasAttrs())
-        for (const Attr *FromAttr : FromD->getAttrs())
-          ToD->addAttr(Importer.Import(FromAttr));
+        for (const Attr *FromAttr : FromD->getAttrs()) {
+          // FIXME: Return of the error here is not possible until
+          // the error is not passed out from GetImportedOrCreateDecl.
+          auto ToAttrOrErr = import(FromAttr);
+          if (ToAttrOrErr)
+            ToD->addAttr(*ToAttrOrErr);
+          else {
+            llvm::errs()
```

Comment:
Shouldn't we issue a proper warning here?

Reply:
What is a "proper warning"? This is really a condition when something could not correctly be imported but probably not an important error (in the old ASTImporter much more similar cases exist without warnings).